search/engine.go

```diff
@@ -316,6 +315,84 @@ func (e *ESEngine) GetSuggestions(ctx context.Context, query Query, preference s
 	return sRes, nil
 }
 
+func FloatOrNil(f *float64) string {
+	if f == nil {
+		return "nil"
+	}
+	return fmt.Sprintf("%.2f", *f)
+}
+
+func LogIfDeb(deb bool, l string) {
```

Comment:
I think it is better to send the Query object as the first param and check if query.Deb is true inside the function body.
This will allow us to set some additional conditions to print logs. it maybe useful in the future.

Reply:
Ack, will fix soon. (adding more logs...)